watertap/unit_models/anaerobic_digestor.py

```diff
@@ -0,0 +1,878 @@
+            self.flowsheet().time, doc="Vapor phase properties", **tmp_dict
+        )
+
+        # ---------------------------------------------------------------------
+        # Check flow basis is compatable
+        # TODO : Could add code to convert flow bases, but not now
+        t_init = self.flowsheet().time.first()
+        if (
+            self.vapor_phase[t_init].get_material_flow_basis()
+            != self.liquid_phase.properties_out[t_init].get_material_flow_basis()
+        ):
+            raise ConfigurationError(
+                f"{self.name} vapor and liquid property packages must use the "
+                f"same material flow basis."
+            )
+
+        self.liquid_phase.add_geometry()
+
+        # Add Ports
+        self.add_inlet_port(name="inlet", block=self.liquid_phase, doc="Liquid feed")
+        self.add_outlet_port(
+            name="liquid_outlet", block=self.liquid_phase, doc="Bottoms stream"
+        )
+        self.add_outlet_port(
+            name="vapor_outlet",
+            block=self.vapor_phase,
+            doc="Vapor stream from reactor",
+        )
+
+        # ---------------------------------------------------------------------
+        # Add unit level constraints
+        # First, need the union and intersection of component lists
+        all_comps = (
+            self.vapor_phase.component_list
+            | self.liquid_phase.properties_out.component_list
+        )
+        common_comps = (
+            self.vapor_phase.component_list
+            & self.liquid_phase.properties_out.component_list
+        )
+
+        # Get units for unit conversion
+        vunits = self.config.vapor_property_package.get_metadata().get_derived_units
+        lunits = self.config.liquid_property_package.get_metadata().get_derived_units
+        flow_basis = self.vapor_phase[t_init].get_material_flow_basis()
+        if flow_basis == MaterialFlowBasis.molar:
+            fb = "flow_mole"
+        elif flow_basis == MaterialFlowBasis.mass:
+            fb = "flow_mass"
+        else:
+            raise ConfigurationError(
+                f"{self.name} AnaerobicDigestor only supports mass or molar "
+                f"basis for MaterialFlowBasis."
+            )
+
+        # Add object references
+        self.volume_liquid = Reference(self.liquid_phase.volume[:])
+
+        self.volume_AD = Var(
+            self.flowsheet().time,
+            initialize=3700,
+            units=lunits("volume"),
+            doc="Volume of the liquid",
+        )
+
+        self.volume_vapor = Var(
+            self.flowsheet().time,
+            initialize=300,
+            units=lunits("volume"),
+            doc="Volume of the gas",
+        )
+
+        self.k_p = Param(
+            initialize=5e4,
+            mutable=True,
+            units=pyunits.m**3 / pyunits.day / pyunits.bar,
+            doc="Component mass concentrations",
```

Comment:
Not sure what the doc description for this should be, but it should be changed.  I'm temporarily calling it "friction parameter" in the documentation since I believe you got it from this paper: http://iwa-mia.org/wp-content/uploads/2019/04/bmadm1_report061127.pdf